lib/tasks/fedramp.rake

```diff
@@ -0,0 +1,65 @@
+require 'io/console'
+
+namespace :rh_cloud do |args|
+  desc 'Register with FedRamp registration API. \
+        Run with INSTANCE=temp, stage, prod to specify an instance to connect to. \
+        Specify org_id=x replace your organization ID with x'
+  task fedramp_register: [:environment] do
+    include ::ForemanRhCloud::CertAuth
+    include ::InsightsCloud::CandlepinCache
+
+    case ENV['INSTANCE']
+    when 'temp' # remove when stage endpoint gets finished out
+      instance = 'https://ephem-tls.outsrights.cc/api/identity/certificate/registrations'
+    when 'stage'
+      instance = 'https://cert.console.stage.openshiftusgov.com/api/identity/certificate/registrations'
+    when 'prod'
+      instance = 'https://cert.console.openshiftusgov.com/api/identity/certificate/registrations'
+    when "/https:/"
+      instance = ENV['INSTANCE']
+    else
+      puts 'ERROR: INSTANCE needs to be specified or is not set to one of the required options.'
+      exit(1)
+    end
+
+    if ENV['org_id'].nil?
+      puts 'ERROR: org_id needs to be specified.'
+      exit(1)
+    else
+      organization = Organization.find_by(id: ENV['org_id'].to_i) # saw this coming in as a string, so making sure it gets passed as an integer.
+    end
+
+    @uid = cp_owner_id(organization)
+
+    puts 'Paste your token, output will be hidden.'
+    @token = STDIN.noecho(&:gets).chomp
+
+    def logger
```

Comment:
Worth moving this one to the beginning of the file to enjoy the ability to log properly